### ISSUE TITLE
Improve the loading of many items

### ISSUE THREAD

**Issue by waliid** [SRGSSR/pillarbox-apple#163]
### Description of the problem

When we try to load a playlist with several items (1000) the UI is blocked.

### Reproducibility

Always

### Steps to reproduce

1. Update _Playlist (URNs, long)_ to contains 1000 items.
2. Open the playlist example previously updated.
3. Try to move forward or the close the player, the UI is blocked.

### Library version

0.1.0

### Operating system

iOS 16.2

### Code sample

```swift
static let longVideos: [Media] = (0...1000).map { _ in
    .urn("urn:rts:video:13588169")
}
```

### Is there an existing issue for this?

- [X] I have searched existing issues and found no similar bug report.

**Comment by defagos** [comment]
Just an idea:

- We should not start metadata retrieval for all items loaded in the player, only those in immediate vicinity. Long jumps would likely add a delay, though.
- We probably don't need to sync the whole `PlayerItem` list with `AVQueuePlayer` item queue. Only a handful of items, possibly only the current one and the next one, are likely needed.

**Comment by defagos** [comment]
This has been implemented in #178.
